integration-tests/tests/src/tests/mixed.ts

```diff
@@ -0,0 +1,309 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2021 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+import { Realm } from "realm";
+import { Decimal128, ObjectId, UUID } from "bson";
+import { expect } from "chai";
+
+const SingleSchema: Realm.ObjectSchema = {
+  name: "mixed",
+  properties: {
+    a: "mixed",
+    b: "mixed",
+    c: "mixed",
+    d: "mixed",
+  },
+};
+
+const VertexSchema: Realm.ObjectSchema = {
+  name: "Vertex",
+  properties: {
+    a: "int",
+    b: "int",
+    c: "int",
+  },
+};
+
+const MixNestedSchema: Realm.ObjectSchema = {
+  name: "Nested",
+  properties: {
+    a: "mixed",
+    b: "mixed",
+    c: "mixed",
+  },
+};
+
+const MixedNullableSchema: Realm.ObjectSchema = {
+  name: "mixed",
+  properties: {
+    nullable: "mixed",
+    nullable_list: "mixed[]",
+  },
+};
+
+const MixedSchema: Realm.ObjectSchema = {
+  name: "MixedClass",
+  properties: { value: "mixed" },
+};
+
+describe("Mixed", () => {
+  describe("primitive values", () => {
+    let realm: Realm;
+    beforeEach(() => {
+      Realm.clearTestState();
+      realm = new Realm({ schema: [SingleSchema] });
+    });
+    afterEach(() => {
+      realm.close();
+    });
+    it("support primitives", function (this: RealmContext) {
```

Comment:
I would like this to use the [`openRealmBeforeEach` hook](https://github.com/realm/realm-js/blob/master/integration-tests/tests/src/hooks/open-realm-before.ts#L43) instead:

```suggestion
    openRealmBeforeEach({ schema: [SingleSchema] });
    it("support primitives", function (this: RealmContext) {
      const { realm } = this;
```

This calls `clearTestState` intenally by default.